### PR TITLE
Refactoring docs to support docsify

### PR DESCRIPTION
**Pull request by sshniro**
### Issues resolved
Docisfy deployed version:
English: https://sshniro.github.io/apisix-docsify/#/
CN: https://sshniro.github.io/apisix-docsify/#/zh-cn/

The PR does not modify the content of the docs. This will be done in the subsequent PRs.

### CODE REVIEW

**Comment by membphis** [comment]
nice PR

**Comment by sshniro** [comment]
This is a list of plugins we can support:
https://docsify.js.org/#/plugins

1. Full-text search
2. Google analytics
3. Copy-code

**Comment by membphis** [comment]
@sshniro you can update your repo first ^_^. https://github.com/sshniro/apisix-docsify

File: doc/zh-cn/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+* [limit-count](plugins/limit-count.md)：基于“固定窗口”的限速实现。
+* [limit-req](plugins/limit-req.md)：基于漏桶原理的请求限速实现。
+* [limit-conn](plugins/limit-conn.md)：限制并发请求（或并发连接）。
+* [proxy-rewrite](plugins/proxy-rewrite.md): 支持自定义修改 proxy 到上游的信息。
+* [prometheus](plugins/prometheus.md)：以 Prometheus 格式导出 APISIX 自身的状态信息，方便被外部 Prometheus 服务抓取。
+* [OpenTracing](plugins/zipkin.md)：支持 Zikpin 和 Apache SkyWalking。
+* [grpc-transcode](plugins/grpc-transcode.md)：REST <--> gRPC 转码。
+* [serverless](plugins/serverless.md)：允许在 APISIX 中的不同阶段动态运行 Lua 代码。
+* [ip-restriction](plugins/ip-restriction.md): IP 黑白名单。
+* [openid-connect](../plugins/oauth.md)
```
Comment (juzhiyuan) [inline doc/zh-cn/README.md:58]: Do we need to keep the position prefix?

File: doc/zh-cn/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+* [redirect](plugins/redirect.md): URI 重定向。
+* [response-rewrite](plugins/response-rewrite.md): 支持自定义修改返回内容的 `status code`、`body`、`headers`。
+* [fault-injection](plugins/fault-injection.md)：故障注入，可以返回指定的响应体、响应码和响应时间，从而提供了不同的失败场景下处理的能力，例如服务失败、服务过载、服务高延时等。
+* [proxy-cache](plugins/proxy-cache.md)：代理缓存插件提供缓存后端响应数据的能力。
+* [proxy-mirror](plugins/proxy-mirror.md)：代理镜像插件提供镜像客户端请求的能力。
+* [udp-logger](../plugins/udp-logger.md): 将请求记录到UDP服务器
+* [tcp-logger](../plugins/tcp-logger.md): 将请求记录到TCP服务器
+* [kafka-logger](plugins/kafka-logger.md): 将请求记录到外部Kafka服务器。
+* [cors](plugins/cors.md): 为你的API启用CORS.
```
Comment (juzhiyuan) [inline doc/zh-cn/README.md:67]: `为你的API启用 CORS。`

File: doc/zh-cn/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+* [grpc-transcode](plugins/grpc-transcode.md)：REST <--> gRPC 转码。
+* [serverless](plugins/serverless.md)：允许在 APISIX 中的不同阶段动态运行 Lua 代码。
+* [ip-restriction](plugins/ip-restriction.md): IP 黑白名单。
+* [openid-connect](../plugins/oauth.md)
+* [redirect](plugins/redirect.md): URI 重定向。
+* [response-rewrite](plugins/response-rewrite.md): 支持自定义修改返回内容的 `status code`、`body`、`headers`。
+* [fault-injection](plugins/fault-injection.md)：故障注入，可以返回指定的响应体、响应码和响应时间，从而提供了不同的失败场景下处理的能力，例如服务失败、服务过载、服务高延时等。
+* [proxy-cache](plugins/proxy-cache.md)：代理缓存插件提供缓存后端响应数据的能力。
+* [proxy-mirror](plugins/proxy-mirror.md)：代理镜像插件提供镜像客户端请求的能力。
+* [udp-logger](../plugins/udp-logger.md): 将请求记录到UDP服务器
```
Comment (juzhiyuan) [inline doc/zh-cn/README.md:64]: ,

**Comment by sshniro** [comment]
> @sshniro you can update your repo first ^_^. https://github.com/sshniro/apisix-docsify

Aw, forgot to update.

https://sshniro.github.io/apisix-docsify/#/README
https://sshniro.github.io/apisix-docsify/#/zh-cn/

Now it's similar to the current PR.

File: doc/zh-cn/README.md (hunk 1 above)
Comment (sshniro) [reply doc/zh-cn/README.md]: Removed

File: doc/zh-cn/README.md (hunk 3 above)
Comment (sshniro) [reply doc/zh-cn/README.md]: Does something need to be added here @juzhiyuan ?

**Comment by membphis** [comment]
it is 404 now:

image

**Comment by sshniro** [comment]
Hi @membphis The reference to the language change should be removed for the main content. The language is switchable via this tab:
image

I did not remove the `Chinese doc` and wanted to remove it in the subsequent PR to easily track changes. And also the Introduction page will be fully rewritten as the current one simply lists the docs, which is already visible in the nav bar,

**Comment by membphis** [comment]
@sshniro 

image

**Comment by sshniro** [comment]
@membphis using absolute path will result in an error.
image

**Comment by sshniro** [comment]
@membphis absolute path results in an error:

image

**Comment by Akayeshmantha** [comment]
this is great work @sshniro

**Comment by membphis** [comment]
> absolute path results in an error

many thanks your reply

**Comment by sshniro** [comment]
Shall we merge this, so I can move forward with the content refactoring?

Also, where are we going to host this? In Github pages? If so we need to change the folder name from `doc` to `docs`

**Comment by membphis** [comment]
merged already. you can continue your job now @sshniro

**Pull request merged**